Vikta-API-testing-sample/src/test/java/com/test/api/user/UserTest.java

```diff
@@ -0,0 +1,446 @@
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import static io.restassured.RestAssured.given;
+import static org.assertj.core.api.Assertions.*;
+
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class UserTest extends BaseTest {
+
+    private final List<Integer> idUserToClean = new ArrayList<>();
+    private static final Logger logger = LogManager.getLogger(CategoryTest.class);
+    private final String idQueryParam = "id";
+
+    @AfterEach
+    void cleanup(TestInfo testInfo) {
+        if (testInfo.getTags().contains("needs-cleanup")) {
+            for(int id : idUserToClean) {
+                given().spec(defaultRequestSpec())
+                        .queryParam(idQueryParam, id)
+                        .when()
+                        .delete(getTestEnvironment().getUserPath());
+            }
+        }
+    }
+
+    @Test
+    public void canGetUserByCorrectId(){
+        List<UserDTO> users = getListOfUsers();
+        final int id = users.get(new Random().nextInt(users.size())).getId();
+
+        Response response = getResponseFromUserById(id);
+
+        UserDTO user = given().spec(defaultRequestSpec())
+                .queryParam(idQueryParam, id)
+                .when()
+                .log().all()
+                .get(getTestEnvironment().getUserPath())
+                .then()
+                .log().all()
+                .spec(defaultResponseSpec())
+                .extract().as(UserDTO.class);
+
+        assertThat(response.getStatusCode()).isEqualTo(200);
+
+        assertThat(user.getId())
+                .as(String.format("Should be user with id %s, but was with %s", id, user.getId()))
+                .isEqualTo(id);
+    }
+
+    @ParameterizedTest
+    @ValueSource(longs = {Long.MIN_VALUE, -20, (long) 0, Long.MAX_VALUE})
+    public void canNotGetUserByWrongId(long id){
+
+        Response response = getResponseFromUserById((int)id);
+
+        assertThat(response.getStatusCode()).isEqualTo(404);
+    }
+
+    @Test
+    public void canListAllUsers(){
+        List<UserDTO> usersList = getListOfUsers();
+
+        assertThat(usersList).as("User list shouldn't be empty").isNotEmpty();
+
+        assertThat(usersList).as("Fields shouldn't be null").allSatisfy(user -> assertThat(user)
+                        .hasNoNullFieldsOrProperties());
+
+        assertThat(usersList).as("Fields shouldn't be empty").allSatisfy(user -> assertThat(user)
+                .usingRecursiveComparison().isNotEqualTo(""));
```

Comment:
here you are trying to check equality between UserDTO object (user) and String (""). If you want to assert that received users don't have empty fields, you can check against UserDTO object that have all String fields set to "".

Reply:
so now is okey?
![Screenshot 2022-02-01 at 3 42 48 PM](https://user-images.githubusercontent.com/94898753/151989612-e8ec8861-942b-4c28-a644-973955c6700f.png)

Reply:
looks good